File: problems/167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (numbers, target) => {
    let l = 0;
    let r = numbers.length - 1;
    while (l < r) {
        if (numbers[l] + numbers[r] === target) return [l + 1, r + 1];
        if (numbers[l] + numbers[r] < target) {
            l ++;
        } else {
            r --;
        }
    }
};

File: problems/167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        while l < r:
            sum = numbers[l] + numbers[r]
            if target == sum:
                return [l + 1, r + 1]
            if sum > target:
                r -= 1
            else:
                l += 1

